spring-thrift-starter/src/main/java/ru/trylogic/spring/boot/thrift/beans/RequestIdLogger.java

```diff
@@ -16,7 +16,7 @@
   private static final String DEFAULT_MDC_KEY = "request_id";
   private static final String X_REQUEST_ID = "x-request-id";
 
-  public String getMDCKey() {
+  public static String getMDCKey() {
```

Comment:
I think it will be better if we extract all async functionality to separated project because there are other async engines such as Hystrix that have similar problems. And I think that multithread problem is not the problem of this library.

So, I think you may use own workaround or additional library till new project will be created.